src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -62,7 +62,8 @@ object FilterIndexRule
                   spark,
                   index,
                   originalPlan,
-                  useBucketSpec = false)
+                  useBucketSpec = HyperspaceConf.useBucketSpecForFilterRule(spark),
+                  useBucketUnionForAppended = false)
```

Comment:
It might be beneficial for later ops which requires bucketing, but just for filter index, we don't need it. 
I think it's better to write a new rule for the cases if needed.

Reply:
Shall we add the comment to the code (since it's not straightforward to understand)?